### PR TITLE
Bug 1209128 - In Music NGA app, search result is not refreshed when m…

### PR DESCRIPTION
**Pull request by justindarc**
…atch is not found

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b5d6ac5f7bd19fad6cc761f0fb2d7ff848208e5b

File: apps/music/js/endpoint.js [hunk 1]
```diff
@@ -384,6 +384,12 @@ function setSongRating(rating, filePath) {
 function search(key, query) {
   return new Promise((resolve) => {
     var results = [];
+
+    if (!query) {
```
Comment (jimporter) [inline apps/music/js/endpoint.js]: Let's move this to `js/db.js`. My goal is to gradually move as much of the DB-related `endpoint.js` logic into `db.js`.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5feb29c0e722935b15a1dfda148f356bdc75dd05

File: apps/music/js/endpoint.js (hunk 1 above)
Comment (justindarc) [inline apps/music/js/endpoint.js]: Good call. Done.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0b3bbaacec67f1bd5590058ba46dc9b34905b960

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5aca064867f9ad04604a19a1828cacf05492d8a1

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=98972eca56ae07b571ae9708907286b2e049f970

**Pull request merged**